Expanded soft_quota.py to include multiple quotas and provide for automation

carlilek opened the pull request:
I've refactored the soft_quota.py code to read a list of quotas from a file and then check all of those quotas, emailing when a quota is exceeded and also creating a csv of quotaname,current_usage. I also added a TERABYTE field and made the default quota definition be in TB. I also changed from KB/MB/GB to KiB/MiB/GiB/TiB. Separated out credentials into a separate file.

carlilek committed:
Expanded soft_quota.py to amke it check for multiple quotas from a quotas.txt file, write current usage to a csv file, as well as separating out the credentials into a credentials.txt file.

carlilek committed:
Added README info and a .gitignore to exclude user settings files.

carlilek committed:
Finished genericising for universal usage

carlilek committed:
Fixed parameter file read paths to allow for cron to run the script.

MichaelSEA commented:
@carlilek apologies that this one sat out there for so long -- my bad.  I don't think I got a notification.  Maybe I did :).  

I will have a look and advise, thanks for PR!

carlilek commented:
No worries!


—Ken


On Jun 16, 2016, at 2:23 PM, Michael Murray <notifications@github.com> wrote:


@carlilek apologies that this one sat out there for so long -- my bad. I don't think I got a notification. Maybe I did :).

I will have a look and advise, thanks for PR!
—
You are receiving this because you were mentioned.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

MichaelSEA commented on quotas_sample.txt:
```diff
@@ -0,0 +1,6 @@
+developers /filestore/departments/developers /mnt/departments/developers 30.0
```
@carlilek this is specific to your installation, can you remove this from the PR please?

carlilek replied:
It’s not specific. That’s a bogus example. 

On Jun 16, 2016, at 2:26 PM, Michael Murray notifications@github.com wrote:

In 
quotas_sample.txt:

> @@ -0,0 +1,6 @@
> +developers /filestore/departments/developers /mnt/departments/developers 30.0

@carlilek this is specific to your installation, can you remove this from the PR please?
—
You are receiving this because you were mentioned.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

MichaelSEA replied:
gotcha.

MichaelSEA commented on credentials_sample.txt:
```diff
@@ -0,0 +1 @@
+cluster1.qumulo.com admin admin 8000
```
@carlilek likewise -- specific to your install.  Please remove from PR, thanks.

carlilek replied:
Also just an example, not a real one. 

On Jun 16, 2016, at 2:27 PM, Michael Murray notifications@github.com wrote:

In 
credentials_sample.txt:

> @@ -0,0 +1 @@
> +cluster1.qumulo.com admin admin 8000

@carlilek likewise -- specific to your install. Please remove from PR, thanks.
—
You are receiving this because you were mentioned.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

MichaelSEA replied:
ah OK I see.... thanks

MichaelSEA commented:
Hi @carlilek  - I don't have rights to carlilek:master so I can't try this out before merging in changes.  Can you create a branch -in this repo- for your changes that I can switch to, verify and then merge?

If you can't do that a zip file via Slack (or other) is fine - thanks!

MichaelSEA commented:
Hi @carlilek - I'd like to get rid of a file that contains credentials.  Not a good security practice.

Instead of keeping credentials in a text file, a better practice is to use environment varials, like the [File Transfer Tool](https://github.com/Qumulo/file_transfer_tool) where I create env vars like so:

```
# file transfer tool
export FTT_CLUSTER=gravytrain
export FTT_CLUSTER_USER=admin
export FTT_CLUSTER_PWD=admin
export FTT_CLUSTER_FOLDER=/users/mmurray/ftt/
```

Can you adapt this PR to work that way as well?

carlilek commented:
I'll have a look at it, sure! I may already be doing something with this in my prod copy...

--Ken


Sent from my <advertising redacted>

On Jun 16, 2016, at 7:57 PM, Michael Murray <notifications@github.com> wrote:




Hi @carlilek - I'd like to get rid of a file that contains credentials. Not a good security practice.
Instead of keeping credentials in a text file, a better practice is to use environment varials, like the
File Transfer Tool where I create env vars like so:
# file transfer tool
export FTT_CLUSTER=gravytrain
export FTT_CLUSTER_USER=admin
export FTT_CLUSTER_PWD=admin
export FTT_CLUSTER_FOLDER=/users/mmurray/ftt/

Can you adapt this PR to work that way as well?
�
You are receiving this because you were mentioned.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

carlilek committed:
modified soft_quota.py to use env variables for creds instead of text file

carlilek committed:
removed credentials_sample.txt

carlilek commented:
I've removed the creds file and added in the env variables for the cluster host, name, and password. Let me know if there's something I missed in the process. 
--Ken

MichaelSEA commented on README.md:
```diff
@@ -1 +1,10 @@
+soft_quota.py will run a check against a Qumulo cluster running Qumulo Core 1.2.9 or later. Requires python 2.7. 
+The script will generate a csv file containing the quota name and the current usage and will email if the quota is exceeded.
+
+Credentials for the Qumulo cluster must be provided in a credentials.txt file. There is a sample file provided. Format is space delimited, 
```
this comment is out of date (and I don't have perms to push to your master, so I can't fix it).  In future you should be able to just create a branch directly on the project in the Qumulo repo, this is a little easier to review :).

carlilek committed:
changed variables to make more clear, fixed readme.md to refer to variables

MichaelSEA commented:
thanks for improvements to this sample @carlilek 👍
